airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py

```diff
@@ -37,21 +39,104 @@ class GoogleAnalyticsSingerSource(BaseSingerSource):
     tap_cmd = "tap-google-analytics"
     tap_name = "Google Analytics API"
     api_error = Exception
+    _custom_reports_schema = {
+        "$schema": "http://json-schema.org/schema#",
+        "type": "array",
+        "items": {
+            "type": "object",
+            "properties": {
+                "name": {"type": "string"},
+                "dimensions": {
+                    "type": "array",
+                    "title": "dimensions",
+                    "items": {
+                        "type": "string",
+                        "enum": [
+                            "ga:browser",
```

Comment:
I believe this shouldn't be an enum as there are custom metrics and custom dimensions that a user could use. For example: https://support.google.com/analytics/answer/2709828?hl=en#reporting

Technically it's an enum, but in order to discover all the available attributes we need to call to a [metadata endpoint](https://developers.google.com/analytics/devguides/config/mgmt/v3/mgmtReference/management/customDimensions/get) and same thing for metrics. Can you create a follow up ticket to make this call and set the JSON schema enum accordingly?

Reply:
@sherifnada - created [ticket](https://github.com/airbytehq/airbyte/issues/2111)